covalent_dispatcher/_cli/service.py

```diff
@@ -320,22 +320,31 @@ def start(
         click.echo(MIGRATION_COMMAND_MSG)
         return ctx.exit(1)
 
-    port = _graceful_start(UI_SRVDIR, UI_PIDFILE, UI_LOGFILE, port, no_cluster, develop)
     no_cluster_flag = "--no-cluster"
     set_config(
         {
-            "user_interface.address": "localhost",
             "user_interface.port": port,
-            "dispatcher.address": "localhost",
             "dispatcher.port": port,
-            "dask": {
-                "mem_per_worker": mem_per_worker or "auto",
-                "threads_per_worker": threads_per_worker or 1,
-                "num_workers": workers or dask.system.CPU_COUNT,
-            },
         }
     )
+    if no_cluster_flag not in sys.argv:
```

Comment:
I think the logic we want is this:
```
if not no_cluster:
    if mem_per_worker != -1:
        set_config({"dask": {"mem_per_worker": mem_per_worker}})
    if threads_per_worker != -1:
        set_config({"dask": {"threads_per_worker": threads_per_worker}})
    if workers != -1:
        set_config({"dask": {"num_workers": workers}})
```

and then for each of those three variables, the click defaults should be set to -1.  The actual defaults that you want in the config should then be modified in the `dask` block inside `covalent/_shared_files/defaults.py`.

Reply:
@wjcunningham7 

I checked the `if no_cluster_flag not in sys.argv:` vs `if not no_cluster:` and they both work.

I just added the updated logic according to your comment. Also note that there's some nuance when the defaults haven't been set yet in the config and the user may or may not have specified some default. I'll probably do some refactoring to reduce some of the code bloat here if you approve the logic